core/src/main/java/com/graphhopper/routing/lm/LandmarkStorage.java

```diff
@@ -897,57 +893,31 @@ static GHPoint createPoint(Graph graph, int nodeId) {
         return new GHPoint(graph.getNodeAccess().getLat(nodeId), graph.getNodeAccess().getLon(nodeId));
     }
 
-    final static class RequireBothDirectionsEdgeFilter implements EdgeFilter {
-
-        private BooleanEncodedValue accessEnc;
-
-        public RequireBothDirectionsEdgeFilter(FlagEncoder flagEncoder) {
-            this.accessEnc = flagEncoder.getAccessEnc();
-        }
-
-        @Override
-        public boolean accept(EdgeIteratorState edgeState) {
-            return edgeState.get(accessEnc) && edgeState.getReverse(accessEnc);
-        }
-    }
-
-    private static class SimpleBlockedEdgesFilter implements EdgeFilter {
-        private final IntSet blockedEdges;
-
-        SimpleBlockedEdgesFilter(IntSet blockedEdges) {
-            this.blockedEdges = blockedEdges;
-        }
-
-        @Override
-        public boolean accept(EdgeIteratorState edgeState) {
-            return !blockedEdges.contains(edgeState.getEdge());
-        }
-    }
-
     private static class BlockedEdgesFilter implements EdgeFilter {
         private final IntHashSet blockedEdges;
-        private final BooleanEncodedValue accessEnc;
+        private final Weighting weighting;
         private final boolean fwd;
         private final boolean bwd;
 
-        public BlockedEdgesFilter(BooleanEncodedValue accessEnc, boolean bwd, boolean fwd, IntHashSet blockedEdges) {
-            this.accessEnc = accessEnc;
+        public BlockedEdgesFilter(Weighting w, boolean bwd, boolean fwd, IntHashSet blockedEdges) {
+            this.weighting = w;
             this.fwd = fwd;
             this.bwd = bwd;
             this.blockedEdges = blockedEdges;
         }
 
         @Override
-        public final boolean accept(EdgeIteratorState iter) {
-            boolean blocked = blockedEdges.contains(iter.getEdge());
+        public final boolean accept(EdgeIteratorState edgeState) {
+            boolean blocked = blockedEdges.contains(edgeState.getEdge());
             if (blocked)
                 return false;
-            return fwd && iter.get(accessEnc) || bwd && iter.getReverse(accessEnc);
+            return fwd && Double.isFinite(weighting.calcEdgeWeightWithAccess(edgeState, false)) ||
+                    bwd && Double.isFinite(weighting.calcEdgeWeightWithAccess(edgeState, true));
```

Comment:
Can we use FiniteWeightFilter here?

Reply:
I currently do not see how. It would only be equal if `fwd` and `bwd` are `true`.

Reply:
Ah yes sorrry nevermind. I just saw this pattern and thought its the same...